katdal/sensordata.py

```diff
@@ -462,6 +462,73 @@ def _safe_linear_interp(xi, yi, x):
     return (1.0 - end_weight) * yi[start] + end_weight * yi[end]
 
 
+_KATCP_DECODER = {'integer': int, 'float': float, 'boolean': lambda x: x == '1',
+                  'lru': str, 'timestamp': float, 'discrete': str,
+                  'address': str, 'string': str}
+
+
+def get_sensor_from_katstore(store, name, start_time, end_time):
+    """Get raw sensor data from katstore (CAM's central sensor database).
+
+    Parameters
+    ----------
+    store : string
+        Hostname / endpoint of katstore webserver speaking katstore REST API
+    name : string
+        Sensor name (the escaped version with underscores)
```

Comment:
What escaping needs to be applied? Or do you mean normalised?

Reply:
The relevant KATCP function is `katcp.resource.escape_name`:

```Python
def escape_name(name):
    """Escape sensor and request names to be valid Python identifiers."""
    return name.replace('.', '_').replace('-', '_')
```

I like "normalised" better but then reverted to the KATCP nomenclature, confusing as it is with the on-the-wire escape mechanism and all.

Reply:
Maybe I'll mention both.